install.sh

```diff
@@ -0,0 +1,171 @@
+echo "Press CTRL+C to abort installation now."
+
+echo -e "${CYAN}Automatically detected: $PROFILE_ROOTDIR${NC}"
+read -e -p "Press ENTER to use this directory, or type in a new one: " newdir
+PROFILE_ROOTDIR="${newdir:-$PROFILE_ROOTDIR}"
+
+if [ ! -d "$PROFILE_ROOTDIR" ]; then
+  echo "ERROR: firefox profile directory could not be found"
+
+  while [ ! -d "$PROFILE_ROOTDIR" ]; do
+    read -p "Enter active root directory found in \"about:profiles\" here: " PROFILE_ROOTDIR;
+
+    if [ ! -d "$PROFILE_ROOTDIR" ]; then
+      echo "invalid directory: specified location does not exist. Try again..."
+    fi
+  done;
+fi
+
+# Check if issued `./installer.sh uninstall`
+if [[ $1 == "uninstall" ]]; then
+  echo -e "${YELLOW}NOTE: This is the final opportunity to abort uninstallation by pressing Ctrl+C${NC}";
+  if ask_question "Delete all files in $PROFILE_ROOTDIR/chrome? [Y/n]: " "y"; then
+    rm -rf $PROFILE_ROOTDIR/chrome;
+  fi;
+
+  reset_all= ask_question "Reset settings in about:config to default? (a backup of user.js and prefs.js will be created if yes) [y/N]: " "n";
+  if $reset_all; then
+    cp "$PROFILE_ROOTDIR/prefs.js" "$PROFILE_ROOTDIR/prefs.js.bak";
+    cp "$PROFILE_ROOTDIR/user.js" "$PROFILE_ROOTDIR/user.js.bak";
+  fi;
+
+  # Download changed user.js from online
+  FETCHED_PREFS=();
+  if ping -c 1 raw.githubusercontent.com &> /dev/null; then
+    while read l;
+      do FETCHED_PREFS+=("$l");
+    done < <(curl -s "https://raw.githubusercontent.com/bmFtZQ/edge-frfox/main/user.js" | grep -E "user_pref\(\"([a-zA-Z0-9.-]+)\",\s*(true|false)\);")
+  else
+    echo -e "${YELLOW}WARNING: Failed to retrieve online user.js prefs. Continuing...${NC}"
+  fi;
+
+  for ((i = 0; i < ${#FETCHED_PREFS[@]}; i++)); do
+    delete_pref "${FETCHED_PREFS[i]}" $reset_all;
+  done;
+
+  ans="y"
+  echo -e "${CYAN}NOTE: enter 'a' to answer 'yes' to all questions.${NC}";
+  for ((i = 0; i < ${#OPTIONALS[@]}; i+=2)); do
+    setting="user_pref(\"${OPTIONALS[i]}\", ${OPTIONALS[i+1]});";
+
+    if [[ ! ${ans,,} =~ ^(all|a)$ ]]; then
+      ans="y"
+      read -e -p "Remove setting $setting from user.js? [Y/a/n]: " in
+      ans="${in:-$ans}";
+
+      
+      if [[ ! ${ans,,} =~ ^(yes|y|all|a)$ ]]; then
+        continue;
+      fi
+    fi;
+
+    delete_pref "$setting" $reset_all;
+  done;
+
+  echo "uninstall complete."
+  exit 0;
+fi
+
+#################
+# INSTALL PHASE #
+#################
+
+echo "Installing..."
+TMP_DIR=$(mktemp -d -t "$PROJECT_NAME.XXXXXX");
+if ! curl -L "https://github.com/bmFtZQ/edge-frfox/archive/refs/heads/main.tar.gz" | tar xz -C $TMP_DIR --strip-components=1; then
+  echo -e "${RED}Installation failed: ${NC}Failed to get github repo tarball and extract it";
+  exit 0;
+fi
+
+echo "Copying theme folder...";
+cp -r $TMP_DIR/chrome $PROFILE_ROOTDIR;
+cat $TMP_DIR/user.js | 
+  while read line; do
+    if ! grep -qF "$line" $PROFILE_ROOTDIR/user.js; then
+      echo "$line" | tee -a $PROFILE_ROOTDIR/user.js >/dev/null;
```

Comment:
It was for testing purposes at the time, since `tee` directs output to the file and `stdout`, therefore very useful for debug logs. It was easier to stop output with `>/dev/null` than it was to change the whole command.

It's better to change it to `echo $contents >> $file` now though.